resolwe/flow/serializers/fields.py

```diff
@@ -11,63 +11,65 @@
 from .descriptor import DescriptorSchemaSerializer
 
 
-class ResolweSlugRelatedField(RelatedField):
+class DictRelatedField(RelatedField):
     """
-    Resolwe specific implementation of SlugRelatedField.
-
-    A read-write field that represents the target of the relationship
-    by a unique combination of 'slug' and 'version' attributes.
-
-    This is a modification of rest_framework.relations.SlugRelatedField.
-    Since slug is not unique, (but combination of slug and version is),
-    we filter objects by slug, by permissions and return object with
+    Field representing the target of the relationship by using dict.
+
+    A read-write field that represents the target of the relationship by
+    a dictionary, where objects are uniquely defined by ``id`` or
+    ``slug`` keys in the dictionary. If ``id`` is provided the uniquness
+    of object is arbitrary. In ``id`` is not in the dict, object is
+    determined by slug. Since multiple objects can have same slug, we
+    filter objects by slug, by permissions and return object with
     highest version.
-
     """
 
     default_error_messages = {
-        'does_not_exist': ('Invalid {model_name} {slug_name} "{value}" - object does not exist.'),
-        'invalid': ('Invalid value.'),
+        'does_not_exist': ('Invalid {model_name} value: {value} - object does not exist.'),
+        'no_data': ('Neither id nor slug is given for field {name}.'),
     }
 
-    def __init__(self, slug_field='slug', **kwargs):
-        """Initialize attributes."""
-        self.slug_field = slug_field
-        super().__init__(**kwargs)
+    def to_representation(self, obj):
+        """Convert to representation."""
+        return {
+            'id': obj.id,
+            'slug': obj.slug,
+        }
 
     def to_internal_value(self, data):
         """Convert to internal value."""
         user = getattr(self.context.get('request'), 'user')
         queryset = self.get_queryset()
         permission = get_full_perm('view', queryset.model)
+
+        if data.get('id', None) is not None:
+            kwargs = {'id': data['id']}
+        elif data.get('slug', None) is not None:
+            kwargs = {'slug': data['slug']}
+        else:
+            self.fail('no_data', name=self.field_name)
+
         try:
             return get_objects_for_user(
                 user,
                 permission,
-                queryset.filter(**{self.slug_field: data}),
+                queryset.filter(**kwargs),
             ).latest()
```

Comment:
Is this OK? I know we set `get_latest_by` in the base model. What if one of the models overrides that, or if we use this field in other use cases at some point. Do we want to be more explicit here and pass `version` to `latest()`?